### DISCUSSION TITLE
Testing ph-ERPA

### DISCUSSION THREAD

**Discussion by gabrielasd** [gabrielasd/pyeom#67] [General]
Hi @PaulWAyers 

Do you think the comparison I attempted in these [notes](https://github.com/gabrielasd/eomee/blob/notebooks/notebook/docs_rpa_particlehole.ipynb) is a valid one that I could use to test our ph-ERPA method? 

I used a 2 electron system with minimal basis so that I could easily select which eigenvalues to include in the evaluation of the TDMs and see the effect it has on the total energy (though these tests are not included in the notes above)

**Comment by PaulWAyers** [comment]
Sorry to respond so late @gabrielasd . 

This reference seems like a great find. I think this is a good test. The issue is, as you say, probably with the fact we don't restrict excitations to preserve spin. I'm still not sure how to very thoroughly test things; I keep going back to a case where all electrons have the same spin, so that the spin-mixing is irrelevant. I'm not sure you'll get a good comparison to the literature that way, though.

It's somewhat alarming how badly the RPA is failing here.

**Reply by gabrielasd** [reply]
Thanks @PaulWAyers  :)

Yes, I don't know any reference I can compare against if I take all electrons with the same spin, but I can still test for that case.

Also, before doing that, I should make sure my implementation of the RPA correction in Tahir's paper is accurate. I have two references with ph-RPA data (likely ones you suggested):
[Scuseria 2013](http://aip.scitation.org/doi/10.1063/1.4820557) has dissociation paths of H2, and LiH
[Tahir 2019](https://link.aps.org/doi/10.1103/PhysRevB.99.195149) has binding curves instead and I'm not sure how to reproduce them, is it E(dimer) - 2*E(atom)? 
and if so what is the atomic specie at association limit for the RHF calculations (Figs 2, 5, 6)? or should I skip those in the comparisons?

thanks!!

**Reply by PaulWAyers** [reply]
Yep, E(dimer) - E(atoms). The atoms should be isolated H atoms for H2, but computed with the same method/basis set. In this case it's just HF in this basis set. For LiH, it should also dissociate to neutral atoms, but the Li calculation may be tricky. It is just a constant shift of the curve, though, so it makes the numbers change but not the delta(E) between two different bond distances.

**Reply by gabrielasd** [reply]
Oh, True! Thanks for making me notice that :) OK I can go from here, thanks!
What is the issue with Li, btw?

**Reply by PaulWAyers** [reply]
Just that for Li, as a doublet, it will be tricky. Correlation energy should be relatively small but I would not expect phRPA to be very good. You can always do it by just dissociating the molecule and (if you can converge the equations) for a very stretched geometry, set that equal to zero to get an (effective) dissociation limit.

**Comment by gabrielasd** [comment]
Contrary to what I said on the notebook the origin of the fluctuations on the E_phrpa(2-RDM) values seen in the last graph, is not ill conditioning of the GEVP matrices, but a potential bug in the generation of the one- and two-electron integrals from which the left-hand-side matrix of the GEVP is built. Therefore the problem is caused by an external function used to run the SCF calculations and  get the integrals and not by the EOM package.

**Reply by PaulWAyers** [reply]
Ugh. So it's to the SCF. These are PySCF integrals? With IOData it should be easy to change software packages.

**Reply by gabrielasd** [reply]
Yes, I'm using a script based on PySCF. I've likely introduced a bug when extracting the electron integrals and transforming them from AO to MO. Also, I checked and this problem (inconsistent energies) happens only with the UHF flavor of that code.
But, yes, I agree I should try interfacing to e.g. gaussian through IOData.
